package.json

```diff
@@ -2,10 +2,13 @@
   "name": "@joincivil/civil-sdk",
   "version": "1.0.0",
   "private": false,
+  "main": "build/static/js/index.js",
+  "types": "build/static/js/index.d.ts",
   "dependencies": {
     "@babel/core": "7.2.2",
     "@commitlint/cli": "^7.5.2",
     "@commitlint/config-conventional": "^7.5.0",
+    "@joincivil/components": "^1.7.5",
```

Comment:
We have an issue here which is that we can only reference published npm versions in package.json, because there's no way to specify using git repo as package when the package we want is in a subdirectory in a monorepo.

It's not an issue during local development (can use yarn link), but the CI build is failing now because of this.

Tons of requests at npm (https://github.com/npm/npm/issues/2974) and yarn (https://github.com/yarnpkg/yarn/issues/4725) for this feature but no dice.

We have three options I can think of:

1. Publish monorepo to npm every time we want to merge a PR in civil-sdk that depends on updates to monorepo package
2. Use https://github.com/ramasilveyra/gitpkg - it takes a subdirectory and creates a package from it under a git tag, which *can* be referenced in package.json - still requires an action but is a single command and more lightweight than npm publish + version increment
3. Update civil-sdk build process with some hack to handle this directly (like checking out latest monorepo and using yarn link)

Reply:
hrm this is a tough one. i think the righest answer is that `components` should be using semantic versioning and published to NPM on changes. that requires setting up though, which I am not sure we have the appetite for right now. its dirty, but i think #3 might be the lightest weight until we determine what our longer term strategy will be. Interested to hear @am9u's thoughts on the matter

Reply:
I agree with @dankins, publishing npm packages when monorepo changes is probably the right way to go as a long-term solution. Perhaps setting up some CircleCI workflow to handle this would be the way to go -- similar to how we do our Production deploys.

But in the interest of prioritizing shipping, a lightweight solution like #3 seems fine for now.

Reply:
Went with #3, a preinstall script. Every yarn/npm install it git pulls latest master monorepo to `./Civil` and rebuilds it. Makes adding a package super slow but the step can be skipped with `--ignore-scripts` flag. Also means you can edit monorepo dependencies directly in that directory instead of yarn link - either way. Clunky but it works! cc @sruddy